File: wxDesigner/samples2.19/scrolled/scrolled_derived.py
#!/bin/env python
#----------------------------------------------------------------------------
# Name:         scrolled_derived.py
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

import wx

# WDR: classes

class MyScrolledWindow(wx.ScrolledWindow):
    def __init__(self, parent, id,
        pos = wx.DefaultPosition, size = wx.DefaultSize,
        style = wx.HSCROLL|wx.VSCROLL ):
        wx.ScrolledWindow.__init__(self, parent, id, pos, size, style)
        
        # WDR: handler declarations for MyScrolledWindow
        wx.EVT_LEFT_DOWN(self, self.OnLeftDown)
        wx.EVT_PAINT(self, self.OnPaint)

    # WDR: methods for MyScrolledWindow

    # WDR: handler implementations for MyScrolledWindow

    def OnLeftDown(self, event):
        dialog = wx.MessageDialog( self, "Left click.", "Test", wx.OK|wx.ICON_INFORMATION )
        dialog.ShowModal()
        dialog.Destroy()

    def OnPaint(self, event):
        dc = wx.PaintDC( self )
        self.PrepareDC( dc )
        dc.DrawLine( 0,0,300,300 )

File: wxDesigner/samples2.19/scrolled/scrolled_derived.pl
#!/usr/bin/perl
#----------------------------------------------------------------------------
# Name:         scrolled_derived.pl
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

use Wx;
use strict;

# WDR: classes

package MyScrolledWindow;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::ScrolledWindow);

use Wx qw(wxICON_INFORMATION wxOK);

sub new {
    my( $class ) = shift;
    my( $this ) = $class->SUPER::new( @_ );
    
    # WDR: handler declarations for MyScrolledWindow
    Wx::Event::EVT_LEFT_DOWN( $this, \&OnLeftDown );
    Wx::Event::EVT_PAINT( $this, \&OnPaint );
    
    $this;
}

# WDR: methods for MyScrolledWindow

# WDR: handler implementations for MyScrolledWindow

sub OnLeftDown {
    my( $this, $event ) = @_;
    
    my( $dialog ) = Wx::MessageDialog->new( $this, "Left click.", "Test", wxOK|wxICON_INFORMATION );
    $dialog->ShowModal();
    $dialog->Destroy();
}

sub OnPaint {
    my( $this, $event ) = @_;
    my( $dc ) = Wx::PaintDC->new( $this );
    $this->PrepareDC( $dc );
    $dc->DrawLine( 0,0,300,300 );
}

